lib/wget.js

```diff
@@ -82,16 +83,18 @@ function download(src, output, options, _parentEvent, redirects) {
                 download(newLocation, output, options, downloader, redirects);
             }
 
-            if (res.statusCode === 200) {
+            if (res.statusCode === 200 || res.statusCode === 206) {
                 downloadedSize = 0;
                 fileSize = Number(res.headers['content-length']);
 
                 // If content length header is not sent, there is no way to determine file size.
                 if (isNaN(fileSize) === true) {
                     fileSize = null;
                 }
+
+                const flags = res.statusCode === 206 ? 'a' : 'w+';
                 writeStream = fs.createWriteStream(output, {
-                    flags: 'w+',
+                    flags,
```

Comment:
If we're writing to separate partial files, then we can prob maintain w+. 

If we want to support writing to sections of the file through range, then this also need to be modified to support a file write offset.